Firestore/core/src/core/pipeline_util.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_CORE_PIPELINE_UTIL_H_
+#define FIRESTORE_CORE_SRC_CORE_PIPELINE_UTIL_H_
+
+#include <memory>
+#include <vector>
+
+#include "Firestore/Protos/nanopb/google/firestore/v1/document.nanopb.h"
+#include "Firestore/core/src/api/expressions.h"
+#include "Firestore/core/src/api/realtime_pipeline.h"
+#include "Firestore/core/src/api/stages.h"
+#include "Firestore/core/src/model/mutable_document.h"
+#include "Firestore/core/src/nanopb/message.h"
```

Comment:
IIUC This header file does not need any of these (23-28) headers. You should be able to remove all of these and only add a "forward declaration" on line 33:

```cpp
class api::EvaluableStage;
```

This makes compilation faster, minimizes coupling of header files, and avoids false circular dependencies in the future. You will have to do the same imports in `pipeline_util.cc` instead.